### PR TITLE
feat: cache OPL when loading from HTTP(S)

### PR DESCRIPTION
**Pull request by alnr**
https://github.com/ory-corp/cloud/issues/5319

### CODE REVIEW

File: internal/driver/config/opl_cache_test.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+	require.NoError(t, err)
+	cw, err := newOPLConfigWatcher(ctx, cfg, ts.URL)
+	require.NoError(t, err)
+	require.Equal(t, 1, hits, "HTTP request made")
+	_, err = cw.GetNamespaceByName(ctx, "User")
+	require.NoError(t, err)
+	_, err = cw.GetNamespaceByName(ctx, "Document")
+	require.NoError(t, err)
+
+	time.Sleep(1 * time.Second) // warm cache
```
Comment (aeneasr) [inline internal/driver/config/opl_cache_test.go:37]: Please use `cache.Wait()` instead, otherwise this will be flaky

**Review by aeneasr — CHANGES_REQUESTED** [review]

File: internal/driver/config/opl_cache_test.go (hunk 1 above)
Comment (alnr) [reply internal/driver/config/opl_cache_test.go]: Uh, brilliant!

File: internal/driver/config/opl_config_namespace_watcher.go [hunk 2]
```diff
@@ -33,7 +36,14 @@ type (
...
 )
 
-var _ namespace.Manager = (*oplConfigWatcher)(nil)
+var (
+	_        namespace.Manager = (*oplConfigWatcher)(nil)
+	cache, _                   = ristretto.NewCache(&ristretto.Config{
+		MaxCost:     20_000_000, // 20 MB max size, each item ca. 10 KB => max 2000 items
+		NumCounters: 20_000,     // max 2000 items => 20000 counters
+		BufferItems: 64,
+	})
```
Comment (aeneasr) [inline internal/driver/config/opl_config_namespace_watcher.go:45]: Ristretto config is a science for itself. I recommend this:

```suggestion
	cache, _                   = ristretto.NewCache(&ristretto.Config{
		NumCounters:        maxItems * 10, // Counters should be 10x max cost (see ristretto docs).
		MaxCost:            maxItems,
		BufferItems:        64,
		Metrics:            true, // consider false
		IgnoreInternalCost: true,
		Cost: func(value interface{}) int64 {
			return 1
		},
 	})
``` 

And then set `maxItems` to `20MB / 10kb = 2048`. Although I'd argue that up to 500MB is fine for this cache.

**Review by zepatrik — APPROVED** [review]

File: internal/driver/config/opl_config_namespace_watcher.go (hunk 2 above)
Comment (alnr) [reply internal/driver/config/opl_config_namespace_watcher.go:45]: Is this better, though? OPL definitions are not all the same size, yet their individual cost is easy to calculate.

LMK.

Comment (aeneasr) [reply internal/driver/config/opl_config_namespace_watcher.go:45]: ok, we can also fine-tune this by checking how this is doing in our tracing

**Pull request merged**